browser/ui/ai_rewriter/ai_rewriter_dialog_delegate.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_AI_REWRITER_AI_REWRITER_DIALOG_DELEGATE_H_
+#define BRAVE_BROWSER_UI_AI_REWRITER_AI_REWRITER_DIALOG_DELEGATE_H_
+
+#include <memory>
+
+#include "base/memory/weak_ptr.h"
+#include "chrome/browser/tab_contents/web_contents_collection.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
```

Comment:
Can these be forward declared?

Reply:
Nice catch! Yeah, I tried a few different ways of observing the contents before settling here :smile: